### PR TITLE
SALTO-3050: Clear secrets from response data before writing to log

### PR DESCRIPTION
**Pull request by shir-reifenberg**
Clear secrets from response data before writing to log

---
_Release Notes_: 
None

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/okta-adapter/src/config.ts [hunk 1]
```diff
@@ -708,14 +708,6 @@ const DEFAULT_TYPE_CUSTOMIZATIONS: OktaApiConfig['types'] = {
       ],
     },
   },
-  IdentityProviderCredentialsClient: {
```
Comment (shir-reifenberg) [inline packages/okta-adapter/src/config.ts:711]: Nor necessary cause it's nested under `credentials` in `Protocol` type which we already omit in line 703

File: packages/okta-adapter/src/client/client.ts [hunk 2]
```diff
@@ -34,6 +36,34 @@ const DEFAULT_PAGE_SIZE: Required<clientUtils.ClientPageSizeConfig> = {
...
 
+/**
+ * Clear response data values might contain secrets, returns a new object
+ */
+const clearSecrets = (
+  object: Values,
+  secretFieldNames: string[]
+): Values => {
+  const SECRET_PLACEHOLER = '<SECRET>'
+  const replaceValuesInplace = (values: Values): void => {
```
Comment (netama) [inline packages/okta-adapter/src/client/client.ts:47]: tiny nit: `replaceValuesInplace` -> `replaceValuesInPlace`

File: packages/okta-adapter/src/client/client.ts [hunk 3]
```diff
@@ -34,6 +36,34 @@ const DEFAULT_PAGE_SIZE: Required<clientUtils.ClientPageSizeConfig> = {
...
+          replaceValuesInplace(values[key])
+        }
+      })
+    }
+    if (Array.isArray(values)) {
+      values.forEach(item => replaceValuesInplace(item))
+    }
+  }
+
+  const clonedObject = _.cloneDeep(object)
```
Comment (netama) [inline packages/okta-adapter/src/client/client.ts:62]: I think we can use `cloneDeepWith` and have the "steps" of `replaceValuesInplace` in the `customizer` and avoid the double traversal/recursion, wdyt?

**Review by netama — COMMENTED** [review]
looks good :) added a suggestion that I _think_ can simplify the logic, if you prefer this approach lmk and I'll approve

**Review by netama — APPROVED** [review]

**Pull request merged**